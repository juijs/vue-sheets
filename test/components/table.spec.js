import { mount } from 'vue-test-utils'
import SheetsTable from '@/components/table'

describe('table.js', () => {
    const wrapper = mount(SheetsTable, {
        propsData: {
            columns: [
                { key: "name", name: "이름", width: "200px" },
                { key: "age", name: "나이" },
                { key: "location", name: "지역" },
            ],
            data: [
                { name: "홍재석", age: "33", location: "대한민국" }
            ],
            templateRow: `<tr>
                <td><!= name !></td>
                <td><!= age !></td>
                <td><!= location !></td>
            </tr>`
        }
    });

    it('renders li for each item in props.items', () => {
        expect(wrapper.vm.data.length).toBe(1);
    })

    it('server render', () => {
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
})