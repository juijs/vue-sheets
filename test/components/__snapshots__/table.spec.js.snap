// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table.js server render 1`] = `
<div
  style="position: relative;"
>
  <table
    class="table classic normal hover"
    style="width: 100%;"
  >
    <thead>
      <tr>
        <th
          title="이름"
          width="200px"
        >
          이름
        </th>
        <th
          title="나이"
          width="auto"
        >
          나이
        </th>
        <th
          title="지역"
          width="auto"
        >
          지역
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        
                
        <td>
          홍재석
        </td>
        
                
        <td>
          33
        </td>
        
                
        <td>
          대한민국
        </td>
        
            
      </tr>
    </tbody>
  </table>
</div>
`;
